packages/salesforcedx-vscode-core/test/vscode-integration/providers/configAggregatorProvider.test.ts

```diff
@@ -0,0 +1,205 @@
+      ConfigAggregatorProvider.prototype as any,
+      'getRootWorkspacePath'
+    );
+    changeCurrentDirectoryToStub = sandbox.stub(
+      ConfigAggregatorProvider.prototype as any,
+      'changeCurrentDirectoryTo'
+    );
+    configAggregatorProvider = ConfigAggregatorProvider.getInstance();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('createConfigAggregator', () => {
+    let getCurrentDirectoryStub: SinonStub;
+    let configAggregatorCreateSpy: SinonSpy;
+
+    beforeEach(() => {
+      configAggregatorProvider = ConfigAggregatorProvider.getInstance();
+      getCurrentDirectoryStub = sandbox.stub(
+        (ConfigAggregatorProvider as any).prototype,
+        'getCurrentDirectory'
+      );
+      configAggregatorCreateSpy = sandbox.spy(ConfigAggregator, 'create');
+    });
+
+    it('should create a ConfigAggregator when outside of a project directory', async () => {
+      // Arrange
+      getCurrentDirectoryStub
+        .onCall(0)
+        .returns(ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE);
+
+      // Act
+      const configAggregator = await (configAggregatorProvider as any).createConfigAggregator();
+
+      // Assert
+      // createConfigAggregator should store the cwd initially,
+      // and check it again after creating the ConfigAggregator
+      // to ensure that the cwd is set back to its original value.
+      expect(getCurrentDirectoryStub.callCount).to.equal(2);
+      // createConfigAggregator should change the process to the
+      // root project workspace path to create the ConfigAggregator,
+      // then change back to the original directory.
+      expect(changeCurrentDirectoryToStub.callCount).to.equal(2);
+      expect(changeCurrentDirectoryToStub.getCall(1).args[0]).to.equal(
+        ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE
+      );
+      expect(configAggregatorCreateSpy.callCount).to.equal(1);
+      expect(configAggregator).to.not.equal(undefined);
+    });
+
+    it('should create a ConfigAggregator when inside of a project directory', async () => {
+      // Arrange
+      getCurrentDirectoryStub.returns(dummyProjectRootWorkspacePath);
+      getRootWorkspacePathStub.returns(dummyProjectRootWorkspacePath);
+
+      // Act
+      const configAggregator = await (configAggregatorProvider as any).createConfigAggregator();
+
+      // Assert
+      // createConfigAggregator should store the cwd initially,
+      // and check it again after creating the ConfigAggregator
+      // to ensure that the cwd is set back to its original value.
+      expect(getCurrentDirectoryStub.callCount).to.equal(2);
+      // Since the stubbed current directory is the same as the root
+      // workspace path, the directory should not have been changed.
+      expect(changeCurrentDirectoryToStub.callCount).to.equal(0);
+      expect(configAggregatorCreateSpy.callCount).to.equal(1);
+      expect(configAggregator).to.not.equal(undefined);
```

Comment:
total nit: I typically do the assertion the returned value first.  If you didn't get back what you expected does it make sense to do the other assertions?